frontend/src/app/App.tsx

```diff
@@ -109,9 +110,29 @@ const App = () => {
 
   if (error) {
     if (appInsights instanceof ApplicationInsights) {
+      // todo: make a type for decoded that inherits from JwtPayload?
+      let decoded: any;
+      let validToken = false;
+      if (accessToken) {
+        try {
+          decoded = jwtDecode(accessToken);
+          validToken = true;
+        } catch (e) {
+          validToken = false;
+          console.error("Failed to decode access token", e);
+        }
+      }
+      const rolesFieldName = `${process.env.REACT_APP_OKTA_SCOPE?.substring(
```

Comment:
this feels hacky/brittle and I want to improve it so PR is back in draft mode - I just pushed so I could test in a lower